### ISSUE TITLE
Tray icon on Linux has changed from white to black

### ISSUE THREAD

**Issue by joeytwiddle** [telegramdesktop/tdesktop#24009]
### Steps to reproduce

1. Use KDE with a dark panel theme, e.g. Breeze
2. Start Telegram and look at the tray icon
4. (You _might_ need some unread messages to reproduce. Probably not though.)

### Expected behaviour

The tray icon should be white, like it has been for the past year.

### Actual behaviour

The tray icon is now black!

![Screenshot_20220202_190655](https://user-images.githubusercontent.com/911799/152143036-7c243db1-259d-47eb-805a-400c27c937d6.png)

I had a look through some commits, and wondered if [this one](https://github.com/desktop-app/lib_lottie/compare/f20776cf6541342c88e540b96e843cadb27016e0...e052821270b913798b65f3a1a292b7ab74107ff6#diff-3e98a18e792acf9339892a5b2fa5a25f7067594fd04c10245f07bb0c5e89d75d) might be related. But that's really just a guess.

### Operating system

Manjaro Linux

### Version of Telegram Desktop

3.5 (started happening 1st / 2nd February)

### Installation source

Static binary from official website

**Comment by Fuchs** [comment]
I'd guess it was https://github.com/telegramdesktop/tdesktop/commit/b65d40a22b3dd3ed9cdb8937543f7b6ea53cec3e

**Comment by herzenschein** [comment]
I experience this issue as well. Also: unread messages are not necessary to reproduce.

**Comment by ratijas** [comment]
FWIW I'm using candy-icons theme, and I don't experience this issue, because the custom icon overrides whatever KDE/Breeze provides.

**Comment by Fuchs** [comment]
Yeah, it's due to the change pointed out, thus the colouring KDE plasma does no longer work. As an ugly workaround until this is fixed in either product, if you use a dark panel you can copy the icons from /usr/share/icons/breeze-dark/status/22/telegram-*     to   /usr/share/icons/breeze/status/22/    (assuming your breeze icons are there, else adapt the path). 
Then delete the plasma cache and reload plasma, this will "fix" it.

**Comment by ilya-fedin** [comment]
This seem to be a Plasma bug. Qt's SNI implementation seem to set both IconPixmap and IconName (what seems reasonable to me - that way if the icon can't be accessible by name due to sandboxing or etc, panel has a chance to fallback to the pixmap), but Plasma prefers IconPixmap and then doesn't color it.

**Comment by ilya-fedin** [comment]
Just in case someone got this wrong: this can't be fixed on Telegram side and I think Qt's behavior is right, so someone should create a KDE bugreport explaining that their tray implementation is bugged (or create a Qt bugreport if you don't agree with my opinion).

@Aokromes I think we don't need this bugreport to be open as it's not a bug in Telegram.

**Comment by akorb** [comment]
I don't have this problem anymore since at least KDE Plasma version 5.23.2 (or KDE Frameworks version 5.91.0, or Qt version 5.15.3). Don't know what contained the fix, but the problem doesn't occur anymore on my machine
